search/collector/topn.go

```diff
@@ -169,13 +186,26 @@ func (hc *TopNCollector) Collect(ctx context.Context, searcher search.Searcher,
 			}
 		}
 
-		err = hc.collectSingle(searchContext, reader, next)
+		err = hc.prepareDocumentMatch(searchContext, reader, next, loadID)
+		if err != nil {
+			break
+		}
+
+		err = dmHandler(next)
 		if err != nil {
 			break
 		}
 
 		next, err = searcher.Next(searchContext)
 	}
+
+	// help finalize/flush the results in case
```

Comment:
should add to the public doc / API comments up above that the handler callback will be invoked with a nil hit, to signal that there aren't any more hits coming?

Reply:
done.